tests/test_approvals.py

```diff
@@ -9,6 +9,129 @@
 # testing this here is more convenient
 from marge.job import CannotMerge, _get_reviewer_names_and_emails
 
+CODEOWNERS = {
+    "content": """
+# This is an example code owners file, lines starting with a `#` will
+# be ignored.
+
+* @default-codeowner @test-user1
+* @test-user1 @ebert
+
+unmatched/* @test5
+"""
+}
+
+# pylint: disable=anomalous-backslash-in-string
+CODEOWNERS_FULL = {
+    "content": """
+# This is an example code owners file, lines starting with a `#` will
+# be ignored.
+
+# app/ @commented-rule
+
+# We can specify a default match using wildcards:
+* @default-codeowner
+
+# Rules defined later in the file take precedence over the rules
+# defined before.
+# This will match all files for which the file name ends in `.rb`
+*.rb @ruby-owner
+
+# Files with a `#` can still be accesssed by escaping the pound sign
+\#file_with_pound.rb @owner-file-with-pound
+
+# Multiple codeowners can be specified, separated by spaces or tabs
+CODEOWNERS @multiple @code @owners
+
+# Both usernames or email addresses can be used to match
+# users. Everything else will be ignored. For example this will
+# specify `@legal` and a user with email `janedoe@gitlab.com` as the
+# owner for the LICENSE file
+LICENSE @legal this_does_not_match janedoe@gitlab.com
+
+# Group names can be used to match groups and nested groups to specify
+# them as owners for a file
+README @group @group/with-nested/subgroup
```

Comment:
groups seem not implemented

Reply:
Good point! I actually just copied the example CODEOWNERS file from the Gitlab documentation to make sure that the file could be parsed correctly for the tests. While support for groups would be amazing I think that's something that should be done in a follow-up MR. I'll remove those references in the test as it's confusing.